tqdm/_tqdm.py

```diff
@@ -39,16 +40,27 @@ class TqdmKeyError(KeyError):
     pass
 
 
-class TqdmDeprecationWarning(DeprecationWarning):
-    # not suppressed if raised
+class TqdmWarning(Warning):
+    """base class for all tqdm warnings.
+
+    Used for non-external-code-breaking errors, such as garbled printing.
+    """
     def __init__(self, msg, fp_write=None, *a, **k):
         if fp_write is not None:
-            fp_write("\nTqdmDeprecationWarning: " + str(msg).rstrip() + '\n')
+            fp_write("\n" + self.__class__.__name__ + ": " + str(msg).rstrip() + '\n')
```

Comment:
https://stackoverflow.com/questions/510972/getting-the-class-name-of-an-instance-in-python/16293038#16293038